engine/apps/api/serializers/organization.py

```diff
@@ -107,3 +108,30 @@ class FastOrganizationSerializer(serializers.ModelSerializer):
     class Meta:
         model = Organization
         fields = ["pk", "name"]
+
+
+class CurrentOrganizationConfigChecksSerializer(serializers.ModelSerializer):
+    is_chatops_connected = serializers.SerializerMethodField()
+    is_integration_chatops_connected = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Organization
+        fields = [
+            "is_chatops_connected",
+            "is_integration_chatops_connected",
+        ]
+
+    def get_is_chatops_connected(self, obj):
+        msteams_backend = get_messaging_backend_from_id("MSTEAMS")
+        return bool(
+            obj.slack_team_identity_id is not None  # slack is connected
+            or obj.telegram_channel.exists()  # telegram is connected
+            or (msteams_backend and msteams_backend.is_configured_for_organization(obj))  # msteams is connected
+        )
+
+    def get_is_integration_chatops_connected(self, obj):
+        return (
+            obj.alert_receive_channels.filter(channel_filters__notify_in_slack=True).exists()
+            or obj.alert_receive_channels.filter(channel_filters__notify_in_telegram=True).exists()
+            or obj.alert_receive_channels.filter(channel_filters__notification_backends__MSTEAMS__enabled=True).exists()
+        )
```

Comment:
I wonder if there's any way we can share the various "chatops connected" logic with this django query?

https://github.com/grafana/oncall/blob/dev/engine/apps/user_management/models/organization.py#L316-L345

Reply:
I see your point, but it seems tricky? I think they have similar but slightly different conditions. Here we aren't really interested in knowing which integration, with less specific restrictions, and we "short-circuit" as soon as we have a satisfying condition (and I would keep the other query as "one query to get them all" for now).